cmake/FindhipSYCL.cmake

```diff
@@ -0,0 +1,37 @@
+find_program(SYCLCC_EXECUTABLE syclcc-clang
+    PATH_SUFFIXES bin)
+
+if(SYCLCC_EXECUTABLE MATCHES "-NOTFOUND")
```

Comment:
In CMake twisted logic, if `SYCLCC_EXECUTABLE` ends with `-NOTFOUND` its already false, so the MATCHES is not required.
Quoting CMake documentation:
```markdown
if(<constant>)

True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number. 
**False** if the constant is 0, OFF, NO, FALSE, N, IGNORE, **NOTFOUND**, the empty string, **or ends in the suffix NOTFOUND**. 
Named boolean constants are case-insensitive. 
If the argument is not one of these specific constants, it is treated as a variable or string and the following signature is used.
```

Reply:
Okay, I can change this if you like.

Reply:
I have corrected this as requested.